src/nationalarchives/components/nested-navigation/nested-navigation.test.ts

```diff
@@ -0,0 +1,160 @@
+import { goToComponent } from '../../../../lib/puppeteer-helpers'
+import { ElementHandle } from 'puppeteer'
+
+import { getExamples } from '../../../../lib/utils'
+
+const getPropertyValue: (item: ElementHandle | null, name: string) => Promise<string> = async (item, name) => {
+  if (item != null) {
+    const property = await item.getProperty(name)
+    return await property.jsonValue() as string
+  }
+  return ''
+}
+
+const getActiveId: () => Promise<string> = async () => {
+  const activeElement = await page.evaluateHandle(() => document.activeElement)
+  const idProperty = await activeElement.getProperty('id')
+  return await idProperty.jsonValue() as string
+}
+
+const goToFirstChild: () => Promise<void> = async () => {
+  await page.waitForSelector('[role="tree"]')
+  await expandRootNode()
+  await page.keyboard.press('ArrowRight')
+}
+
+const goToSecondChild: () => Promise<void> = async () => {
+  await goToFirstChild()
+  await page.keyboard.press('ArrowDown')
+}
+
+const isRootNodeExpanded: () => Promise<boolean> = async () => {
+  const nodeItem = await page.$('[role="tree"] .govuk-tna-tree__node-item')
+  const expanded = await getPropertyValue(nodeItem, 'ariaExpanded')
+  return expanded === 'true'
+}
+
+const expandRootNode: () => Promise<void> = async () => {
+  const isExpanded = await isRootNodeExpanded()
+  if (!isExpanded) {
+    await page.keyboard.press('Tab')
+    await page.keyboard.press('Tab')
+    await page.keyboard.press('ArrowRight')
+  } else {
+    await page.keyboard.press('Tab')
+    await page.keyboard.press('Tab')
+  }
+}
+
+interface NestedNavigationData {
+  name: string
+  type: string
+  id: string
+  children: NestedNavigationData[]
+}
+
+describe('Nested navigation', () => {
```

Comment:
Should there be a test for when the parent node is selected, al child nodes are selected?

Reply:
Yeah, that makes sense, I've added one in.